pkg/model/components/context.go

```diff
@@ -177,8 +178,7 @@ func Image(component string, architecture string, clusterSpec *kops.ClusterSpec,
 	baseURL := clusterSpec.KubernetesVersion
 	baseURL = strings.TrimSuffix(baseURL, "/")
 
-	// TODO path.Join here?
-	tagURL := baseURL + "/bin/linux/" + architecture + "/" + component + ".docker_tag"
+	tagURL := path.Join(baseURL, "/bin/linux/", architecture, component) + ".docker_tag"
```

Comment:
Shouldn't "/bin/linux/" omit the leading and trailing slashes?

Why is the ".docker_tag" outside the path.Join? If `component` is the empty string, these two are not equivalent.

Reply:
https://play.golang.org/p/EK7ti5lYdSp

Reply:
see https://play.golang.org/p/zrbeKMcGKg0

Reply:
it seems that path.join will result `"memfs://v1.10.0-download"`  to  `memfs:/v1.10.0-download/`,it will fail. It's not reasonable to use this here.@johngmyers what do you say?

Reply:
You'd have to parse the url and use `path.Join` on the path component only.

I'm skeptical of the value of using `path.Join` in this instance. It doesn't seem to be solving a problem that simple concatenation doesn't.

Reply:
agree